src/bp/ui-studio/src/web/views/FlowBuilder/diagram/nodes_v2/SaySomethingNode/index.tsx

```diff
@@ -0,0 +1,98 @@
+import classnames from 'classnames'
+import _ from 'lodash'
+import React, { Component } from 'react'
+import { AbstractNodeFactory, DiagramEngine } from 'storm-react-diagrams'
+
+import { BaseNodeModel } from '../../nodes/BaseNodeModel'
+import { StandardPortWidget } from '../../nodes/Ports'
+import style from '../style.scss'
+import { showHeader } from '../utils'
+
+import SayNodeContent from './SayNodeContent'
+
+export class SaySomethingWidget extends Component<{
+  node: SaySomethingNodeModel
+  diagramEngine: any
+  contentLang?: string
+  defaultLanguage?: string
+}> {
+  forceRefresh = false
+  currentNode
+
+  constructor(props) {
+    super(props)
+
+    this.currentNode = { ...props.node }
+  }
+
+  render() {
+    const { node } = this.props
+
+    if (JSON.stringify({ ...this.currentNode.formData }) !== JSON.stringify({ ...node.formData })) {
+      this.forceRefresh = !this.forceRefresh
+      this.currentNode = { ...node }
+    }
+
+    return (
+      <div
+        onClick={() => this.props.diagramEngine.flowBuilder.props.switchFlowNode(this.props.node.id)}
+        className={classnames(style.baseNode, style.nodeSaySomething, { [style.highlightedNode]: node.isHighlighted })}
+      >
+        {showHeader({ nodeType: 'Say', nodeName: node.name, isStartNode: node.isStartNode })}
+        <div className={style.content}>
+          <SayNodeContent forceRefresh={this.forceRefresh} node={node} />
+        </div>
+        <div className={style.ports}>
+          <StandardPortWidget name="in" node={node} className={style.in} />
+          <StandardPortWidget name="out0" node={node} className={style.out} />
+        </div>
+      </div>
+    )
+  }
+}
+
+export class SaySomethingNodeModel extends BaseNodeModel {
+  public contentType
+  public formData = {}
+
+  constructor({
+    id,
+    x,
+    y,
+    name,
+    onEnter = [],
+    next = [],
+    contentType,
+    formData = {},
+    isStartNode = false,
+    isHighlighted = false
+  }) {
+    super('say_something', id)
+    this.setData({ name, onEnter, next, contentType, formData, isStartNode, isHighlighted })
+
+    this.x = this.oldX = x
+    this.y = this.oldY = y
+  }
+
+  setData({ contentType, formData, ...data }: any) {
+    this.contentType = contentType
+    this.formData = formData
+
+    super.setData(data as any)
+  }
+}
+
+export class SaySomethingWidgetFactory extends AbstractNodeFactory {
+  constructor() {
+    super('say_something')
+  }
+
+  generateReactWidget(diagramEngine: DiagramEngine, node: SaySomethingNodeModel) {
+    return <SaySomethingWidget node={node} diagramEngine={diagramEngine} />
+  }
```

Comment:
I played with it a little and this would work to update the node content when the selected language changes: 

```js
generateReactWidget(diagramEngine: DiagramEngine, node: SaySomethingNodeModel) {
    const Wrapped = withLanguage(SaySomethingWidget)
    return <Wrapped node={node} diagramEngine={diagramEngine} />
  }
```

Reply:
This updates the child components a lot more frequently than the forceRefresh I implemented